### PR TITLE
Add project list command to show list of projects in a company

### PR DESCRIPTION
**Pull request by shilgapira**
## Description
- Add `descope project list` command to show list of projects in a company

## Must
- [X] Tests
- [X] Documentation (if applicable)

### CODE REVIEW

File: project/cmd.go [hunk 1]
```diff
@@ -19,6 +19,10 @@ func AddCommands(parent *cobra.Command, group *cobra.Group) {
 	project := shared.MakeGroupCommand(group, "project", "Commands for managing projects")
 	parent.AddCommand(project)
 
+	shared.AddCommand(project, List, "list", "Lists all projects with a company management key", func(cmd *cobra.Command) {
```
Comment (dorsha) [inline project/cmd.go:22]: Do we need the `with a company management key`?

Comment (shilgapira) [reply project/cmd.go:22]: it won't work with a key limited to specific projects

Comment (shilgapira) [reply project/cmd.go]: removed

Comment (dorsha) [reply project/cmd.go]: yes but this description is a bit weird and not clear.
Maybe:
`Lists all projects (using a company management key)`
or something like that.

**Review by dorsha — APPROVED** [review]

**Pull request merged**